### PR TITLE
rename parameters in add_images_weighted

### LINKED ISSUE #369: Add back _index names to functions
To minimize backwards compatibility breaking changes, I propose to rename the `row`, `column`, `plane`, `slice` parameters back to `..._index` as it was in the _prototype:
* `set_plane`
* `set_row`
* `set_column`
* `copy_slice`
* `copy_horizontal_slice`
* `copy_vertical_slice`

### PR DESCRIPTION
**Pull request by haesleinhuepf**
Here I'm renaming numeric parameters of some functions to improve backwards-compatibility with the prototype

closes #369
partially solves clEsperanto/pyclesperanto#260

### CODE REVIEW

**Comment by StRigaud** [comment]
Sorry in advance for the pre-commit thing 😅

**Comment by haesleinhuepf** [comment]
@StRigaud This is ready for review. Feel free to merge if it's ok!

File: clic/include/tier1.hpp [hunk 1]
```diff
@@ -47,8 +47,8 @@ add_images_weighted_func(const Device::Pointer & device,
                          const Array::Pointer &  src0,
                          const Array::Pointer &  src1,
                          Array::Pointer          dst,
-                         float                   factor0,
-                         float                   factor1) -> Array::Pointer;
+                         float                   factor1,
+                         float                   factor2) -> Array::Pointer;
```
Comment (StRigaud) [inline clic/include/tier1.hpp:51]: I know it is to fit the `prototype` but the input index being 0 and 1, and the factor index being 1 and 2 triggers my OCD.

Wouldn't we want to also use this effort to correct possible wrong doing in the API developpment during the `prototype` ( i know I am annoying :wink: )

File: clic/src/tier1/add_images_weighted.cpp [hunk 2]
```diff
@@ -13,13 +13,13 @@ add_images_weighted_func(const Device::Pointer & device,
                          const Array::Pointer &  src0,
                          const Array::Pointer &  src1,
                          Array::Pointer          dst,
-                         float                   factor0,
-                         float                   factor1) -> Array::Pointer
+                         float                   factor1,
+                         float                   factor2) -> Array::Pointer
```
Comment (StRigaud) [inline clic/src/tier1/add_images_weighted.cpp:17]: same as first comment

**Comment by StRigaud** [comment]
This will require to update some of the tests in pyclesperanto because of the parameter name change but otherwise, should be ok.

Just the factor indexing that annoys me a bit... the rest i can live with it :upside_down_face:

File: clic/include/tier1.hpp (hunk 1 above)
Comment (haesleinhuepf) [reply clic/include/tier1.hpp:51]: The documentation is also commonly "First image to process"

**Pull request merged**